plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsBlobContainer.java

```diff
@@ -230,14 +243,26 @@ private void writeToPath(
         EnumSet<CreateFlag> createFlags
     ) throws IOException {
         final byte[] buffer = new byte[blobSize < bufferSize ? Math.toIntExact(blobSize) : bufferSize];
-        try (FSDataOutputStream stream = fileContext.create(blobPath, createFlags, CreateOpts.bufferSize(buffer.length))) {
+        Options.CreateOpts[] createOpts = getCreationOptions(CreateOpts.bufferSize(buffer.length));
+        try (FSDataOutputStream stream = fileContext.create(blobPath, createFlags, createOpts)) {
             int bytesRead;
             while ((bytesRead = inputStream.read(buffer)) != -1) {
                 stream.write(buffer, 0, bytesRead);
             }
         }
     }
 
+    private Options.CreateOpts[] getCreationOptions(Options.CreateOpts... baseOpts) {
```

Comment:
Nit: The variable name `baseOpts` doesn't seem right. These are more like `overrideOpts` since they're provided on a call-by-call basis. The replication factor value is something I'd consider to be a base option since it's stable throughout the whole life of the object.

Reply:
I've removed that method altogether for now.